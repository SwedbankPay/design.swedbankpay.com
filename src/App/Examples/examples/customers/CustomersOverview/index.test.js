import React from "react";
import { shallow } from "enzyme";

import CustomersOverview from "./index";

describe("Examples: CustomersOverview", () => {
    const mockSetCustomerIndex = jest.fn();
    const mockCustomersList = [
        {
            id: "c1",
            firstName: "Sven",
            lastName: "Svensson",
            email: "sven.svensson@email.com",
            phone: "+47 *** ** ***",
            location: "Visby",
            status: "default"
        },
        {
            id: "c2",
            firstName: "Anna",
            lastName: "Svensson",
            email: "anna.svensson@email.com",
            phone: "+47 *** ** ***",
            location: "Stockholm",
            status: "success"
        },
        {
            id: "c3",
            firstName: "Ola",
            lastName: "Nordmann",
            email: "ola.nordmann@email.com",
            phone: "+47 *** ** ***",
            location: "Oslo",
            status: "success"
        }
    ];
    const mockStatusText = {
        default: "New",
        success: "Delivered",
        warning: "Late",
        danger: "Missing"
    };

    it("is defined", () => {
        expect(CustomersOverview).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<CustomersOverview setCustomerIndex={mockSetCustomerIndex} customersList={mockCustomersList} statusText={mockStatusText} />);

        expect(wrapper).toMatchSnapshot();
    });
});
