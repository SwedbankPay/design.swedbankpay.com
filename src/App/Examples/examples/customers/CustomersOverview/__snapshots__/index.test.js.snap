// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Examples: CustomersOverview renders 1`] = `
<div
  className="container"
>
  <div
    className="panel panel-brand"
  >
    <header>
      <div
        className="row"
      >
        <div
          className="col-lg-4"
        >
          <h2
            className="panel-title"
          >
            Customers
          </h2>
        </div>
        <div
          className="col-lg-8"
        >
          <div
            className="input-group"
          >
            

            <span
              className="input-group-addon"
            >
              

              <i
                aria-hidden="true"
                className="material-icons"
              >
                search
              </i>
            </span>
            

            <input
              className="form-control"
              onChange={[Function]}
              type="text"
              value=""
            />
            

            <button
              className="btn btn-primary btn-outline"
              disabled={true}
              onClick={[Function]}
              type="button"
            >
              

              <i
                aria-hidden="true"
                className="material-icons"
              >
                close
              </i>
              

            </button>
            

          </div>
        </div>
      </div>
    </header>
    <div
      className="panel-table"
    >
      <table
        className="table table-hover table-striped"
      >
        <thead>
          <tr>
            <th
              scope="col"
            >
              ID
            </th>
            <th
              className="d-none d-sm-table-cell"
              scope="col"
            >
              First name
            </th>
            <th
              className="d-none d-sm-table-cell"
              scope="col"
            >
              Last name
            </th>
            <th
              scope="col"
            >
              Location
            </th>
            <th
              className="d-none d-lg-table-cell"
              scope="col"
            >
              Delivery status
            </th>
            <th
              className="d-lg-none"
              scope="col"
            />
          </tr>
        </thead>
        <tbody>
          <tr
            key="c1"
            onClick={[Function]}
          >
            <th
              scope="row"
            >
              c1
            </th>
            <td
              className="d-none d-sm-table-cell"
            >
              Sven
            </td>
            <td
              className="d-none d-sm-table-cell"
            >
              Svensson
            </td>
            <td>
              Visby
            </td>
            <td
              className="d-none d-lg-table-cell"
            >
              <span
                className="status status-default"
              >
                New
              </span>
            </td>
            <td
              className="d-lg-none"
            >
              <span
                className="status status-default pl-0"
              />
            </td>
          </tr>
          <tr
            key="c2"
            onClick={[Function]}
          >
            <th
              scope="row"
            >
              c2
            </th>
            <td
              className="d-none d-sm-table-cell"
            >
              Anna
            </td>
            <td
              className="d-none d-sm-table-cell"
            >
              Svensson
            </td>
            <td>
              Stockholm
            </td>
            <td
              className="d-none d-lg-table-cell"
            >
              <span
                className="status status-success"
              >
                Delivered
              </span>
            </td>
            <td
              className="d-lg-none"
            >
              <span
                className="status status-success pl-0"
              />
            </td>
          </tr>
          <tr
            key="c3"
            onClick={[Function]}
          >
            <th
              scope="row"
            >
              c3
            </th>
            <td
              className="d-none d-sm-table-cell"
            >
              Ola
            </td>
            <td
              className="d-none d-sm-table-cell"
            >
              Nordmann
            </td>
            <td>
              Oslo
            </td>
            <td
              className="d-none d-lg-table-cell"
            >
              <span
                className="status status-success"
              >
                Delivered
              </span>
            </td>
            <td
              className="d-lg-none"
            >
              <span
                className="status status-success pl-0"
              />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <Pagination
      arrows={true}
      farArrows={true}
      text="1 of 123"
    />
  </div>
</div>
`;
